### PR TITLE
Docs for dolt_workspace_ tables

### PR DESCRIPTION
**Pull request by macneale4**

### CODE REVIEW

**Review by max-hoffman — APPROVED** [review]
lgtm, some version of "workspace is the union of staged and working set diffs" would have saved me some time understanding the term

File: packages/dolt/content/reference/sql/version-control/dolt-system-tables.md [hunk 1]
```diff
@@ -1108,6 +1109,71 @@ WHERE staged=false;
 +------------+--------+-----------+
 ```
 
+## `dolt_workspace_$TABLENAME`
+
+This system table shows you which rows have been changed in your workspace and if they are staged.
```
Comment (zachmu) [inline packages/dolt/content/reference/sql/version-control/dolt-system-tables.md:1114]: If we're going to use the term workspace, we should have docs for the concept, similar to working set:

https://docs.dolthub.com/concepts/dolt/git/working-set

File: packages/dolt/content/reference/sql/version-control/dolt-system-tables.md [hunk 2]
```diff
@@ -1108,6 +1109,71 @@ WHERE staged=false;
...
+Any table listed in `dolt_status` table will have a non-empty corresponding `dolt_workspace_$TABLENAME`
+table. Changes listed are all relative to the HEAD of the current branch.
+
+These tables can be modified in order to update what changes are staged for commit.
+[Workspace review](https://www.dolthub.com/blog/2024-08-16-workspace-review/)
+
+### Schema
+
+Each row in the `dolt_workspace_$TABLENAME` corresponds to a single row update in the table.
```
Comment (zachmu) [inline packages/dolt/content/reference/sql/version-control/dolt-system-tables.md:1128]: Should mention the source table schema before this workspace table schema.

File: packages/dolt/content/reference/sql/version-control/dolt-system-tables.md [hunk 3]
```diff
@@ -1108,6 +1109,71 @@ WHERE staged=false;
...
+call to [`dolt_commit()`](dolt-sql-procedures.md#dolt_commit). Changes which have `staged = FALSE` are present in your
+workspace which means all queries in your session contain them but they will not be recorded
+in the event that [`dolt_commit()`](dolt-sql-procedures.md#dolt_commit) is executed.
+
+There are two ways you can alter the state of your workspace using these tables.
+1) The `staged` column can be toggled for any row. If changing from false to true, the row values will be moved
+to staging. If there are already staged changes for that row, they will be overwritten. If changing from true to
+false, the row values will be unstaged. If there are other changes in the workspace for that row, the workspace
+change will be preserved and the staged change will be dropped.
+2) Any row which has `staged = FALSE` can be deleted. This will result in reverting the change in question.
```
Comment (zachmu) [inline packages/dolt/content/reference/sql/version-control/dolt-system-tables.md]: The change to the row?

**Review by zachmu — APPROVED** [review]
LGTM, some suggestions

**Pull request merged**